x/mongo/driver/auth/oidc.go

```diff
@@ -0,0 +1,261 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+	"go.mongodb.org/mongo-driver/x/mongo/driver"
+)
+
+const MongoDBOIDC = "MONGODB-OIDC"
+const tokenResourceProp = "TOKEN_RESOURCE"
+const environmentProp = "ENVIRONMENT"
+const allowedHostsProp = "ALLOWED_HOSTS"
+
+const azureEnvironmentValue = "azure"
+const gcpEnvironmentValue = "gcp"
+
+const apiVersion = 1
+const invalidateSleepTimeout = 100 * time.Millisecond
+const machineCallbackTimeout = 60 * time.Second
+
```

Comment:
I just ported over all the constants, some of these will not be used until other tickets like Human flow